BigQuery/src/Model.php

```diff
@@ -0,0 +1,229 @@
+     */
+    public function __construct(
+        ConnectionInterface $connection,
+        $id,
+        $datasetId,
+        $projectId,
+        array $info = []
+    ) {
+        $this->connection = $connection;
+        $this->identity = [
+            'modelId' => $id,
+            'datasetId' => $datasetId,
+            'projectId' => $projectId
+        ];
+
+        $this->info = $info;
+    }
+
+    /**
+     * Retrieve's the model's details. If no model data is cached, a network request
+     * will be made to retrieve it.
+     *
+     * Example:
+     * ```
+     * $info = $model->info();
+     * echo $info['modelType'];
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/rest/v2/models#resource Model resource documentation.
+     * @param array $options
+     * @return array
+     */
+    public function info(array $options = [])
+    {
+        if (!$this->info) {
+            $this->reload($options);
+        }
+
+        return $this->info;
+    }
+
+    /**
+     * Triggers a network request to reload the model's details.
+     *
+     * Example:
+     * ```
+     * $model->reload();
+     * $info = $model->info();
+     * echo $info['modelType'];
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/rest/v2/models/get Models get API documentation.
+     *
+     * @param array $options
+     * @return array
+     */
+    public function reload(array $options = [])
+    {
+        return $this->info = $this->connection->getModel($options + $this->identity);
+    }
+
+    /**
+     * Retrieves the model's ID.
+     *
+     * Example:
+     * ```
+     *  echo $model->id();
+     * ```
+     *
+     * @return string
+     */
+    public function id()
+    {
+        return $this->identity['modelId'];
+    }
+
+    /**
+     * Retrieves the model's identity.
+     *
+     * An identity provides a description of a resource that is nested in nature.
+     *
+     * Example:
+     * ```
+     * echo $model->identity()['modelId'];
+     * ```
+     * @return array
+     */
+    public function identity()
+    {
+        return $this->identity;
+    }
+
+    /**
+     * Delete the model.
+     *
+     * Please note that by default the library will not attempt to retry this
+     * call on your behalf.
+     *
+     * Example:
+     * ```
+     * $model->delete();
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/rest/v2/models/delete Models delete API documentation.
+     *
+     * @param array $options [optional]
+     */
+    public function delete(array $options = [])
+    {
+        $this->connection->deleteModel(
+            $options
+            + ['retries' => 0]
+            + $this->identity
+        );
+    }
+
+    /**
+     * Check whether or not the model exists
+     *
+     * Example:
+     * ```
+     * echo $model->exists();
+     * ```
+     * @return bool
+     */
+    public function exists()
```

Comment:
Any method which makes a request needs to include `$options` and pass it to the connection. (Add it to the docblock too) :)